src/main/kotlin/com/colivery/serviceaping/rest/v1/services/AdminRestService.kt

```diff
@@ -0,0 +1,39 @@
+package com.colivery.serviceaping.rest.v1.services
+
+import com.colivery.serviceaping.mapping.toUserResource
+import com.colivery.serviceaping.persistence.entity.UserEntity
+import com.colivery.serviceaping.persistence.repository.UserRepository
+import com.colivery.serviceaping.rest.v1.dto.user.PatchUserAdminDto
+import com.colivery.serviceaping.rest.v1.resources.UserResource
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.server.reactive.ServerHttpResponse
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+import reactor.core.publisher.Mono
+import javax.transaction.Transactional
+import javax.validation.Valid
+
+@RestController
+@PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
Could we use `@RolesAllowed` or `@Secured` instead? Would make it more clear and introduces less magic (no method calls inside the string parameter).

Reply:
I couldn't get them to work.. only the @Preauthorize =/
I think the SecurityContext is created differently (different bean) with WebFlux and the other annotations are not registering themselves with the correct one...